src/pkg/services/m365/api/lists.go

```diff
@@ -251,42 +251,49 @@ func (c Lists) PostList(
 		BySiteId(siteID).
 		Lists().
 		Post(ctx, newList, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "creating list")
+	}
+
+	listItems, err := c.PostListItems(ctx, siteID, restoredList)
+	if err == nil {
+		restoredList.SetItems(listItems)
+		return restoredList, nil
+	}
 
-	return restoredList, graph.Wrap(ctx, err, "creating list").OrNil()
+	// rollback list creation
+	err = c.DeleteList(ctx, siteID, ptr.Val(restoredList.GetId()))
```

Comment:
can you please add a `TODO` comment here to say that we need to investigate the behavior we want?